src/components/admin/settings/general/Update.vue

```diff
@@ -33,15 +34,25 @@ const [seoTitle] = defineField("seo.title");
 const [seoDescription] = defineField("seo.description");
 const [seoTwitter] = defineField("seo.twitter");
 
+// Reactive array to store featured links
+const featuredLinks = ref<{ name: string; url: string }[]>([{ name: '', url: '' }]);
+
+const addFeaturedLink = () => {
+  featuredLinks.value = [...featuredLinks.value, { name: '', url: '' }];
+};
+
+// Initialize fields with data from settings
 watchEffect(() => {
   if (generalSettings.data.value) {
-    // Setting defaults
     const gs = generalSettings.data.value;
 
     organizationName.value = gs.organization.name;
     organizationDescription.value = gs.organization.description;
     organizationLocation.value = gs.organization.location;
-    organizationLinks.value = gs.organization.links;
+    featuredLinks.value = (gs.organization.links || []).map(link => ({
+      name: link.name || link.url,
+      url: link.url
+    }));
```

Comment:
you don't need this, just take care of it while saving. also, we can guarantee array initialisation by including in SEED FN (see src/server/tasks/seed-database) so need to do `|| []`

Reply:
featuredLinks.value = gs.organization.links.map(link => ({name: link.name || link.url, url: link.url }));

So i removed || [] . But the + button is not able to add new rows. Should i keep (gs.organization.links || []).map ???

Reply:
I am saying you will have to fix that by modifying seed.

Replace below with 
```
 { 
       key: 'organizationConfig', 
       value: '{ links: [] }', 
     }, 
```
https://github.com/profilecity/vidur/blob/6dcba1ef59d5d152055c20b7046d12d7070abb41/src/server/utils/seeds.ts#L11-L22

Then clear your database and restart server. This will initialise the array.

Reply:
updated bro!